### PR TITLE
refactor: update `blas/ext/base/scusumors`

### LINKED ISSUE #1519: [RFC]: refactor `blas/ext/base/scusumors` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/scusumors`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/scusumors) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by rxbryan**
updates blas/ext/base/scusumors to follow current project conventions

## Related Issues

> Does this pull request have any related issues?
#788
#1152 

This pull request:

-   fixes #1519 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md [hunk 1]
```diff
@@ -65,18 +65,15 @@ The function has the following parameters:
 -   **y**: output [`Float32Array`][@stdlib/array/float32].
 -   **strideY**: index increment for `y`.
 
-The `N` and `stride` parameters determine which elements in `x` and `y` are accessed at runtime. For example, to compute the cumulative sum of every other element in `x`,
+The `N` and `stride` parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the cumulative sum of every other element in `x`,
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md]:
```suggestion
The `N` and stride parameters determine which elements in the strided arrays are accessed at runtime. For example, to compute the cumulative sum of every other element in `x`,
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md [hunk 2]
```diff
@@ -158,20 +149,17 @@ scusumors.ndarray( N, 0.0, x, 2, 1, y, -1, y.length-1 );
...
 ```javascript
-var randu = require( '@stdlib/random/base/randu' );
-var round = require( '@stdlib/math/base/special/round' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var Float32Array = require( '@stdlib/array/float32' );
 var scusumors = require( '@stdlib/blas/ext/base/scusumors' );
 
 var y;
 var x;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md:158]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md [hunk 3]
```diff
@@ -158,20 +149,17 @@ scusumors.ndarray( N, 0.0, x, 2, 1, y, -1, y.length-1 );
...
-var i;
 
-x = new Float32Array( 10 );
+x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
 y = new Float32Array( x.length );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
var y = new Float32Array( x.length );
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md (hunk 2 above)
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/docs/repl.txt [hunk 4]
```diff
@@ -44,24 +44,24 @@
     > {{alias}}( x.length, 0.0, x, 1, y, 1 )
     <Float32Array>[ 1.0, -1.0, 1.0 ]
 
-    // Using `N` and `stride` parameters:
+    // Using `N` and stride parameters:
     > x = new {{alias:@stdlib/array/float32}}( [ -2.0, 1.0, 1.0, -5.0, 2.0, -1.0 ] );
     > y = new {{alias:@stdlib/array/float32}}( x.length );
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
-    > {{alias}}( N, 0.0, x, 2, y, 2 )
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/examples/index.js [hunk 5]
```diff
@@ -18,20 +18,17 @@
...
 
-var randu = require( '@stdlib/random/base/randu' );
-var round = require( '@stdlib/math/base/special/round' );
 var Float32Array = require( '@stdlib/array/float32' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' ).factory;
+var filledarrayBy = require( '@stdlib/array/filled-by' );
 var scusumors = require( './../lib' );
 
 var y;
 var x;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/examples/index.js]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/examples/index.js [hunk 6]
```diff
@@ -18,20 +18,17 @@
...
-var i;
 
-x = new Float32Array( 10 );
+x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
 y = new Float32Array( x.length );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
var y = new Float32Array( x.length );
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/manifest.json [hunk 7]
```diff
@@ -1,40 +1,46 @@
...
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/napi/export",
+        "@stdlib/napi/argv",
+        "@stdlib/napi/argv-float",
+        "@stdlib/napi/argv-int64",
+        "@stdlib/napi/argv-strided-float32array"
+      ]
+    }
+  ]
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/manifest.json:74]: We need to add configuration for benchmark and examples here now, see e.g. 
https://github.com/stdlib-js/stdlib/blob/afb14b3920c53ac9f95ed89e745ed98999672ef9/lib/node_modules/%40stdlib/blas/ext/base/dfill/manifest.json#L49

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "stdlib/blas/ext/base/scusumors.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_float.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_float32array.h"
+#include <node_api.h>
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+*
+* -   When called from JavaScript, the function expects six arguments:
+*
+*     -   `N`: number of indexed elements
+*     -   `sum`: initial sum
+*     -   `X`: input array
+*     -   `strideX`: `X` stride length
+*     -   `Y`: output array
+*     -   `strideY`: `Y` stride length
+*/
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c]:
```suggestion
/**
* Receives JavaScript callback invocation data.
*
* @private
* @param env    environment under which the function is invoked
* @param info   callback data
* @return       Node-API value
*/
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, sum, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, Y, N, strideY, argv, 4 );
+
+	stdlib_strided_scusumors( N, (float)sum, (float *)X, strideX, (float *)Y, strideY );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/src/addon.c]:
```suggestion
	stdlib_strided_scusumors( N, sum, X, strideX, Y, strideY );
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.ndarray.js [hunk 11]
```diff
@@ -141,7 +140,7 @@ tape( 'the function returns a reference to the output array', function test( t )
 	t.end();
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `y` unchanged', function test( t ) {
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output unchanged', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.ndarray.js]:
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output array unchanged', function test( t ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.ndarray.native.js [hunk 12]
```diff
@@ -150,7 +149,7 @@ tape( 'the function returns a reference to the output array', opts, function tes
 	t.end();
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `y` unchanged', opts, function test( t ) {
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output unchanged', opts, function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.ndarray.native.js]:
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output array unchanged', opts, function test( t ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.scusumors.js [hunk 13]
```diff
@@ -141,7 +140,7 @@ tape( 'the function returns a reference to the output array', function test( t )
 	t.end();
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `y` unchanged', function test( t ) {
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output unchanged', function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.scusumors.js]:
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output array unchanged', function test( t ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.scusumors.native.js [hunk 14]
```diff
@@ -150,7 +149,7 @@ tape( 'the function returns a reference to the output array', opts, function tes
 	t.end();
 });
 
-tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `y` unchanged', opts, function test( t ) {
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output unchanged', opts, function test( t ) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/scusumors/test/test.scusumors.native.js]:
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns the output array unchanged', opts, function test( t ) {
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Almost there; just a few changes needed. Thanks!

File: lib/node_modules/@stdlib/blas/ext/base/scusumors/manifest.json (hunk 7 above)
Comment (rxbryan) [reply lib/node_modules/@stdlib/blas/ext/base/scusumors/manifest.json:74]: I've implemented the changes.

**Comment by rxbryan** [comment]
Hi @Planeshifter, I have implemented the changes

**Review by Planeshifter — APPROVED** [review]
LGTM! Will merge after CI has passed.

**Pull request merged**